### PR TITLE
Added docker tool for visual studio code.

### PR DESCRIPTION
**Pull request by peacememories**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

This flatpak adds a docker executable to the vscode flatpak app. It is directly modeled after [com.visualstudio.code.tool.podman](https://github.com/flathub/com.visualstudio.code.tool.podman).
Regarding the above points:
* I am not an upstream contributor to either Docker or Visual Studio Code, but it does not seem like this would concern either of them, since it is not a full packaging of an app. If I should ask for permission to publish this, please say so.
* I do not own the domain `code.visualstudio.com`. Instead, this domain is used to follow convention and make the tool better discoverable, similar to the podman tool

There are no patches or other modifications used.
This is my first time building a flatpak, so if i inadvertently broke anything or forgot to include something, please tell me and I'll be happy to fix it and learn something.

### CODE REVIEW

File: com.visualstudio.code.tool.docker.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  - '*.la'
+  - /share/man
+modules:
+  - name: docker
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 docker -t ${FLATPAK_DEST}/bin/
+    sources:
+      - type: archive
+        url: https://download.docker.com/linux/static/stable/x86_64/docker-20.10.12.tgz
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.visualstudio.code.tool.docker.yml:23]: but why not building from source? If I'm not mistaken Docker also supports aarch64. So does the vscode package.

File: com.visualstudio.code.tool.docker.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="runtime">
+  <id>com.visualstudio.code.tool.docker</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>Apache-2.0</project_license>
+  <name>Podman</name>
+  <summary>Docker is a platform built for developers to build and run applications.</summary>
+  <url type="homepage">https://docker.com/</url>
+  <releases>
+    <release version="20.10.12" date="2022-03-29"/>
```
Comment (hfiguiere) [inline com.visualstudio.code.tool.docker.metainfo.xml:10]: shouldn't date be "2020-10-12"

Comment (peacememories) [reply com.visualstudio.code.tool.docker.metainfo.xml:10]: So it should be the release date of the upstream software I guess? Makes sense, will change

File: com.visualstudio.code.tool.docker.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (peacememories) [reply com.visualstudio.code.tool.docker.yml:23]: I guess I can look into building from source. I honestly just took the path of least resistance to get this working for me^^ Will see if I can change that

File: com.visualstudio.code.tool.docker.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="runtime">
+  <id>com.visualstudio.code.tool.docker</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>Apache-2.0</project_license>
+  <name>Podman</name>
```
[the remark this answers is not in the document]
Comment (peacememories) [reply com.visualstudio.code.tool.docker.metainfo.xml:6]: Yep, missed that when "borrowing" the podman definition, oops

File: com.visualstudio.code.tool.docker.metainfo.xml (hunk 2 above)
Comment (peacememories) [reply com.visualstudio.code.tool.docker.metainfo.xml:10]: The actual release date of 20.10.12 seems to be 2021-12-13, if I'm reading the release notes correctly. Gonna use that. (Don't ask me how that naming convention works^^')

**Comment by peacememories** [comment]
@hfiguiere sorry if I'm taking too much of your time, but I have a question. In the podman package the option `branch: 21.08` is set. That version matches the sdk version it depends on. Is that how that should always work?
I'm looking for a way to maybe publish multiple versions of the package for different docker versions so people can choose, but I haven't found that in the doc yet (I'm sure it's there, I just don't know where to look^^')

**Comment by hfiguiere** [comment]
bot, build com.visualstudio.code.tool.docker

File: flathub.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "skip-icons-check": true,
+    "automerge-flathubbot-prs": true,
```
Comment (hfiguiere) [inline flathub.json:3]: there is no auto checking, so this is not useful at all.

Comment (peacememories) [reply flathub.json]: Reverted this change

File: com.visualstudio.code.tool.docker.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (peacememories) [reply com.visualstudio.code.tool.docker.yml:23]: For now I would keep this as a binary install if it's okay, and look into building from source later on

**Comment by peacememories** [comment]
Hmm, can I poke the build bot as well? Let me try...

**Comment by peacememories** [comment]
bot, build com.visualstudio.code.tool.docker

File: flathub.json (hunk 4 above)
Comment (peacememories) [reply flathub.json]: It seems the icon check skip was indeed important

**Comment by peacememories** [comment]
bot, build com.visualstudio.code.tool.docker

**Comment by peacememories** [comment]
Seems to be working okay. Should I change anything else?

**Comment by gasinvein** [comment]
Why exactly do you need the real `docker` binary at all? I'm not against adding it, but maybe its user-case already can be covered with Podman, given the REST API is largely compatible.

**Pull request closed without merging**